src/stk/itk_bridge/itk_io.cpp

```diff
@@ -0,0 +1,304 @@
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::CHAR: {
+        using PixelType = itk::Vector<char, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::USHORT: {
+        using PixelType = itk::Vector<unsigned short, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::SHORT: {
+        using PixelType = itk::Vector<short, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::UINT: {
+        using PixelType = itk::Vector<unsigned int, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::INT: {
+        using PixelType = itk::Vector<int, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::ULONG: {
+        using PixelType = itk::Vector<unsigned long, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::LONG: {
+        using PixelType = itk::Vector<long, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::ULONGLONG: {
+        using PixelType = itk::Vector<unsigned long long, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::LONGLONG: {
+        using PixelType = itk::Vector<long long, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::FLOAT: {
+        using PixelType = itk::Vector<float, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::DOUBLE: {
+        using PixelType = itk::Vector<double, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::UNKNOWNCOMPONENTTYPE:
+    default:
+        FATAL() << "Cannot read image '" << &file_name << "': "
+                << "unknown component type";
+    }
+}
+
+
+/*!
+ * \brief Read a vectorial image of unspecified type with ITK.
+ */
+template<unsigned int Dimension>
+stk::Volume read_vector_image(
+        const std::string& file_name,
+        const itk::ImageIOBase::IOComponentType component_type,
+        const unsigned int components
+        )
+{
+    switch (components)
+    {
+    case 1:
+        return read_vector_image<Dimension, 1>(file_name, component_type);
```

Comment:
Here I am treating scalar images as vectors with one component, to simplify the code a lot. In ITK they are two distinct types, but since it is only a temporary object it should not matter and the final result on file should be the same (or, at least, I hope so).